File: core/src/ru/alexkulikov/firstfame/background/MountainDrawer.java
package ru.alexkulikov.firstfame.background;

import com.badlogic.gdx.scenes.scene2d.Stage;

public class MountainDrawer {
    private Mountain leftMountain;
    private Mountain centralMountain;
    private Mountain rightMountain;

    private float mainOffset = 0;

    public MountainDrawer(Stage stage) {
        leftMountain = new Mountain();
        centralMountain = new Mountain();
        rightMountain = new Mountain();

        stage.addActor(leftMountain);
        stage.addActor(centralMountain);
        stage.addActor(rightMountain);
    }

    public void initialize() {
        leftMountain.setSpriteX(0);
        centralMountain.setSpriteX(leftMountain.getSpriteOffset());
        rightMountain.setSpriteX(centralMountain.getSpriteOffset());
        mainOffset = 0;
    }

    public void update(float camX) {
        leftMountain.setSpriteX((0 - camX/2) + mainOffset);
        centralMountain.setSpriteX(leftMountain.getSpriteOffset());
        rightMountain.setSpriteX(centralMountain.getSpriteOffset());

        if (leftMountain.getSpriteOffset() < 0) {
            change();
        }
    }

    public void change() {
        Mountain buffer = leftMountain;
        mainOffset += leftMountain.getSpriteWidth();

        leftMountain = centralMountain;
        centralMountain = rightMountain;
        rightMountain = buffer;

        rightMountain.setSpriteX(centralMountain.getSpriteOffset());
    }
}

File: core/src/ru/alexkulikov/firstfame/background/MountainDrawer.kt
package ru.alexkulikov.firstfame.background

import com.badlogic.gdx.scenes.scene2d.Stage

class MountainDrawer(stage: Stage) {
    private var leftMountain = Mountain()
    private var centralMountain = Mountain()
    private var rightMountain = Mountain()

    private var mainOffset = 0f

    init {
        stage.addActor(leftMountain)
        stage.addActor(centralMountain)
        stage.addActor(rightMountain)
    }

    fun initialize() {
        centralMountain.setSpriteX(0f)
        leftMountain.setSpriteX(centralMountain.sprite.x - leftMountain.sprite.width)
        rightMountain.setSpriteX(centralMountain.spriteOffset)
        mainOffset = 0f
    }

    fun update(camX: Float) {
        centralMountain.setSpriteX(0 - camX / 2 + mainOffset)
        leftMountain.setSpriteX(centralMountain.sprite.x - leftMountain.sprite.width)
        rightMountain.setSpriteX(centralMountain.spriteOffset)

        if (leftMountain.spriteOffset < -leftMountain.sprite.width / 2) {
            changeLeft()
        } else if (rightMountain.sprite.x > rightMountain.sprite.width * 1.5) {
            changeRight()
        }
    }

    private fun changeLeft() {
        val buffer = leftMountain
        mainOffset += leftMountain.sprite.width

        leftMountain = centralMountain
        centralMountain = rightMountain
        rightMountain = buffer

        rightMountain.setSpriteX(centralMountain.spriteOffset)
    }

    private fun changeRight() {
        val buffer = rightMountain
        mainOffset -= rightMountain.sprite.width

        rightMountain = centralMountain
        centralMountain = leftMountain
        leftMountain = buffer

        leftMountain.setSpriteX(centralMountain.spriteOffset)
    }
}
